src/common/Finisher.cc

```diff
@@ -77,15 +80,13 @@ void *Finisher::finisher_thread_entry()
 	  c->complete(ls_rval.front().second);
 	  ls_rval.pop_front();
 	}
-	if (logger) {
-	  logger->dec(l_finisher_queue_len);
-	  end = ceph_clock_now();
-	  logger->tinc(l_finisher_complete_lat, end - start);
-	  start = end;
-        }
       }
       ldout(cct, 10) << "finisher_thread done with " << ls << dendl;
       ls.clear();
+      if (logger) {
+	logger->dec(l_finisher_queue_len, count);
+	logger->tinc(l_finisher_complete_lat, ceph_clock_now() - start);
```

Comment:
This changes what complete_lat means.. it's now a single latency measurement for the whole batch, whereas before it was a latency measurement for each completion from teh start of the batch.  Which doesn't really make any sense, so this is good!